assembl/static2/js/app/components/debate/common/sentiments.jsx

```diff
@@ -50,60 +51,62 @@ export const Sentiment = ({
+          client
+            .mutate(
+              isSelected
+                ? {
+                  mutation: deleteSentimentMutation,
+                  variables: { postId: postId },
+                  optimisticResponse: {
+                    deleteSentiment: {
+                      post: {
+                        id: postId,
+                        sentimentCounts: {
+                          like: likeCount - (mySentiment === sentiment.type ? 1 : 0),
+                          disagree: disagreeCount - (mySentiment === sentiment.type ? 1 : 0),
+                          dontUnderstand: dontUnderstandCount - (mySentiment === sentiment.type ? 1 : 0),
+                          moreInfo: moreInfoCount - (mySentiment === sentiment.type ? 1 : 0),
+                          __typename: 'SentimentCounts'
+                        },
+                        mySentiment: null,
+                        __typename: 'Post'
                       },
-                      mySentiment: null,
-                      __typename: 'Post'
-                    },
-                    __typename: 'DeleteSentiment'
+                      __typename: 'DeleteSentiment'
+                    }
                   }
                 }
-              }
-              : {
-                mutation: addSentimentMutation,
-                variables: { postId: postId, type: sentiment.type },
-                optimisticResponse: {
-                  addSentiment: {
-                    post: {
-                      id: postId,
-                      sentimentCounts: {
-                        like: sentiment.camelType === 'like' ? likeCount + 1 : likeCount - (mySentiment === 'LIKE' ? 1 : 0),
-                        disagree:
-                            sentiment.camelType === 'disagree'
-                              ? disagreeCount + 1
-                              : disagreeCount - (mySentiment === 'DISAGREE' ? 1 : 0),
-                        dontUnderstand:
-                            sentiment.camelType === 'dontUnderstand'
-                              ? dontUnderstandCount + 1
-                              : dontUnderstandCount - (mySentiment === 'DONT_UNDERSTAND' ? 1 : 0),
-                        moreInfo:
-                            sentiment.camelType === 'moreInfo'
-                              ? moreInfoCount + 1
-                              : moreInfoCount - (mySentiment === 'MORE_INFO' ? 1 : 0),
-                        __typename: 'SentimentCounts'
+                : {
+                  mutation: addSentimentMutation,
+                  variables: { postId: postId, type: sentiment.type },
+                  optimisticResponse: {
+                    addSentiment: {
+                      post: {
+                        id: postId,
+                        sentimentCounts: {
+                          like: sentiment.camelType === 'like' ? likeCount + 1 : likeCount - (mySentiment === 'LIKE' ? 1 : 0),
```

Comment:
refactoring: add a new function in the component body:
```
const getNewCount = (count: number, sentimentType: string): number=>{
  return sentiment.type === sentimentType ? count + 1 : count - (mySentiment === sentimentType ? 1 : 0)
}
```
and then:

`like: getNewCount(likeCount, SENTIMENTS.like)`